### ISSUE TITLE
about the uncondition car shape performance

### ISSUE THREAD

**Issue by HANYUEZHAN** [JohnRomanelis/SPVD#12]
What is the relationship among 1-NN-CD-acc，1-NN-CD-acc_f, 1-NN-CD-acc_t, and which specific CD and EMD values (1-NN-CD-acc，1-NN-CD-acc_f, 1-NN-CD-acc_t) are reported in Table I? Thanks for your great work and for open-sourcing this project!

<img width="628" height="586" alt="Image" src="https://github.com/user-attachments/assets/6732308c-7fa9-4e0b-a7e3-51b058e2a209" />

**Comment by JohnRomanelis** [comment]
Hey @HANYUEZHAN ! Thanks for your kind words 😁.

Short answer:

The **1-NN-CD-acc** is reported in the paper.

Long answer:

- To compute the 1‑NN metric we use two sets of point clouds:
  - reference_set: real (ground-truth) point clouds
  - sample_set: generated point clouds
- For each point cloud in both sets we find its nearest neighbor (using Chamfer distance or EMD).
  - 1-NN-CD-acc_f is the fraction of sample (generated) point clouds whose nearest neighbor by Chamfer distance is also a sample.
  - 1-NN-CD-acc_t is the fraction of reference (real) point clouds whose nearest neighbor by Chamfer distance is also a reference.
- The 1‑NN metric measures the fraction of point clouds whose nearest neighbor has the same label; The ideal value is 50% (0.5) when the selection of the neighbor subset becomes random. Reporting acc_t and acc_f separately gives more insight into the per-set behaviour.

Feel free to ask if you have any further questions!

**Comment by HANYUEZHAN** [comment]
Thanks for your response！ I now have a clearer understanding of the model evaluation metrics. Your answer has been a huge help.

**Comment by HANYUEZHAN** [comment]
Dear JohnRomanelis,
I hope you're doing well. I'm trying to reproduce the SPVD-S results for Car (CD and EMD) in Table 1, and would like to confirm the configurations:
Testing:
model = SPVUnet(voxel_size=0.1, nfs=(32, 64, 128, 256),num_layers=1, attn_chans=8, attn_start=3)
Training:
lr = 0.001, epochs = 4000
Same model architecture as above
model = SPVUnet(voxel_size=0.1, nfs=(32, 64, 128, 256),num_layers=1, attn_chans=8, attn_start=3)
tr_dataset.set_voxel_size(1e-5)
tr_dataset.set_noise_params(beta_max=0.02)

te_dataset.set_voxel_size(1e-5)
te_dataset.set_noise_params(beta_max=0.02)
Are these the correct settings used in your paper? Any additional details would be greatly appreciated.
Thank you for your help!
Best regards,
HANYUEZHAHN

**Comment by JohnRomanelis** [comment]
Hey @HANYUEZHAN !

I’ve edited your previous messages to remove the email history since you replied via mail - just a heads-up 🙂

Yes, your understanding is correct. I’m not entirely sure about the learning rate, but I believe that’s the value we used. If you notice the loss turning into `nan` during training, try reducing the learning rate by half. Starting with `0.001` is a good baseline, though.

Also, when setting the noise parameters, make sure you’re using the correct values for `beta_min` and `n_steps`:

```python
tr_dataset.set_noise_params(beta_min=0.0001, beta_max=0.02, n_steps=1000)
```

Hope this helps! If you’d like to skip retraining, you can load the pretrained checkpoints I’ve provided.

Let me know if you run into any other issues with your implementation.

**Comment by HANYUEZHAN** [comment]
Hi  JohnRomanelis,
Thank you so much for your prompt reply and the detailed information! This is very helpful.
I just checked my code and found that during testing, I'm using the correct noise parameters:
`ddpm_sched = DDPMSparseSchedulerGPU(n_steps=1000, beta_min=0.0001, beta_max=0.02)`
However, I realized that during training, I was only setting beta_max=0.02 without specifying beta_min and n_steps. This might explain why my results don't match the paper yet.
I'll retrain the model with the correct parameters you provided:
`tr_dataset.set_noise_params(beta_min=0.0001, beta_max=0.02, n_steps=1000)`
Additionally, I'd like to ask: can I use the chair.pt pretrained model under Generation/SPVD_S to reproduce the Chair results in Table 1? I want to verify if my testing setup is correct by comparing with multiple categories.
Thanks again for your help and quick response!
Best regards,
HANYUEZHAN

**Comment by JohnRomanelis** [comment]
Yes, the pretrained checkpoints are the ones used in the paper, so feel free to use them.
Keep in mind that there’s some stochasticity in the testing process, so your results might not match the paper exactly, but they should be very close.

I also double-checked that the default values for `set_noise_params` are indeed the ones I mentioned earlier, so that’s likely not the issue.
If possible, try evaluating one of the provided checkpoints to confirm that your evaluation code is working as expected.

**Comment by HANYUEZHAN** [comment]
Thank you so much for your help! I just tested the pre-trained model, and it's very close to what's described in the paper. Thank you again! Your research is truly remarkable and has been immensely helpful to me.



At 2025-10-22 22:52:01, "Ioannis Romanelis" ***@***.***> wrote:

JohnRomanelis left a comment (JohnRomanelis/SPVD#12)

Yes, the pretrained checkpoints are the ones used in the paper, so feel free to use them.
Keep in mind that there’s some stochasticity in the testing process, so your results might not match the paper exactly, but they should be very close.

I also double-checked that the default values for set_noise_params are indeed the ones I mentioned earlier, so that’s likely not the issue.
If possible, try evaluating one of the provided checkpoints to confirm that your evaluation code is working as expected.
